src/pam_oauth2_device.cpp

```diff
@@ -373,20 +404,28 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc, cons
     Config config;
     DeviceAuthResponse device_auth_response;
     Userinfo userinfo;
+    int ret;
```

Comment:
Probably it's safer to initialize the varialble, e.g.
```suggestion
    int ret = PAM_SYSTEM_ERR;
```

Reply:
ret was completely removed.